From b49ab1cd739bfcf18c0a130e0808ae1f9a222bff Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Thu, 10 May 2018 23:47:13 +0200
Subject: [PATCH] mmc: sdhci-msm: Use max-frequency property

Use the passed max-frequency property from sdhci node instead of using 400MHz as default.
That behaviour breaks stuff
---
 drivers/mmc/host/sdhci-msm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-msm.c b/drivers/mmc/host/sdhci-msm.c
index 92c483ec6..38be89128 100644
--- a/drivers/mmc/host/sdhci-msm.c
+++ b/drivers/mmc/host/sdhci-msm.c
@@ -1190,7 +1190,7 @@ static int sdhci_msm_probe(struct platform_device *pdev)
 	}
 
 	/* Vote for maximum clock rate for maximum performance */
-	ret = clk_set_rate(msm_host->clk, INT_MAX);
+	ret = clk_set_rate(mmc->f_max);
 	if (ret)
 		dev_warn(&pdev->dev, "core clock boost failed\n");
 
